File: practice/algorithms/greedy/two_arrays/TwoArrays.java
import java.util.*;

public class TwoArrays {

    private static class Descending implements Comparator<Integer> {
        @Override
        public int compare(Integer i, Integer j) {
            return j - i;
        }
    }

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int tests = stdin.nextInt();
        for(int i = 0; i < tests; i++) {
            int n = stdin.nextInt();
            int sum = stdin.nextInt();
            Integer[] array1 = new Integer[n];
            Integer[] array2 = new Integer[n];
            for(int j = 0; j < n; j++){
                array1[j] = stdin.nextInt();
            }
            for(int j = 0; j < n; j++){
                array2[j] = stdin.nextInt();
            }
            Arrays.sort(array1);
            Arrays.sort(array2, new Descending());
            System.out.println(isCorrect(array1, array2, sum));
        }
        stdin.close();
    }

    private static String isCorrect(Integer[] array1, Integer[] array2, int sum) {
        for(int i = 0; i < array1.length; i++) {
            if(array1[i] + array2[i] < sum) {
                return "NO";
            }
        }
        return "YES";
    }
}

File: practice/algorithms/greedy/two_arrays/TwoArrays.scala
import scala.io.Source

object TwoArrays extends App {
  val console = Source.stdin.bufferedReader()
  val tests = console.readLine().toInt
  val descendingSort = (i: Int, j: Int) => i > j
  (0 until tests).foreach(i => {
    val line = console.readLine()
    val numbers = line.split(" ").map(_.toInt)
    val n = numbers(0)
    val sum = numbers(1)
    val array1 = console.readLine().split(" ").map(_.toInt)
    val array2 = console.readLine().split(" ").map(_.toInt)
    val sortedArray1 = array1.sorted
    val sortedArray2 = array2.sortWith(descendingSort)
    println(prove(sortedArray1, sortedArray2, sum))
  })

  def prove(arr1: Array[Int], arr2: Array[Int], sum: Int): String = {
    val correct = arr1.indices.forall(i => {arr1(i) + arr2(i) >= sum})
    if(correct)
      "YES"
    else
      "NO"
  }
}